### ISSUE TITLE
cleanup deprecated CAPI annotations

### ISSUE THREAD

**Issue by enxebre** [kubernetes/autoscaler#4453]
**Which component are you using?**:
CAPI provider is using some deprecated annotations. We should clean them up.
e.g https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go#L32-L34

**What version of the component are you using?**:

Component version:

**What k8s version are you using (`kubectl version`)?**:

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version

</pre></details>

**What environment is this in?**:

**What did you expect to happen?**:

**What happened instead?**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Comment by elmiko** [comment]
hi @sandipanpanda , there are some code changes that will need to be done.

to begin, look at the code that is highlighted in the original issue description:
https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go#L32-L34

```go
const (
	// deprecatedMachineDeleteAnnotationKey should not be removed until minimum cluster-api support is v1alpha3
	deprecatedMachineDeleteAnnotationKey = "cluster.k8s.io/delete-machine"
	// TODO: determine what currently relies on deprecatedMachineAnnotationKey to determine when it can be removed
	deprecatedMachineAnnotationKey = "cluster.k8s.io/machine"
```

the issue here is that these 2 constants: `deprecatedMachineDeleteAnnotationKey` and `deprecatedMachineAnnotationKey` have been fully deprecated from the upstream [cluster-api project](https://github.com/kubernetes-sigs/cluster-api). they should be removed from the code here.

you will need to check in the source files in the `cluster-autoscaler/cloudprovider/clusterapi` directory to see where these values are being used, and then modify the code so that we can remove the constant variables. They are used in a couple unit tests and also in the unstructured module, it should be fairly straightforward to determine how they can be removed but if you have trouble feel free to ask questions either here or on the kubernetes slack in the #sig-autoscaling channel. you can also ping me directly with questions, my handle is the same on slack.

**Comment by elmiko** [comment]
this is closed by #4463
/close
